types/check.ts

```diff
@@ -764,3 +766,86 @@ websocket1.addEventListener('message', evt => {
 	assert<MessageEvent>(evt);
 	assert<string>(evt.data);
 });
+
+/**
+ * WORKTOP/MODULES
+ */
+
+const worker1: ModuleWorker = {
+	async fetch(req, env, ctx) {
+		assert<Request>(req);
+		assert<IncomingCloudflareProperties>(req.cf);
+
+		assert<Bindings>(env);
+
+		// @ts-expect-error
+		let foobar = env.foobar;
```

Comment:
This is an error because even though `env.foobar` _might_ exist, it wasn't actually defined in the `ModuleWorker` generic. So here, `env` is effectively types as a placeholder holder for "bindings will exist on me .... but I won't let you use any unless you define it!"